### ISSUE TITLE
� Symbol at the end of every comment

### ISSUE THREAD

**Issue by ghost** [InstaPy/InstaPy#1740]
## Expected Behavior
No question mark box.

## Current Behavior
A question mark box � appears at the end of every comment.

I applied the solution proposed in #1411. 
in `comment_util.py`
I have encoded `rand_comment`
`rand_comment = (choice(comments).format(username)).encode('utf-8')`
And I removed it from `logger.info`
`logger.info("--> Commented: {}".format(rand_comment))`

I doesn't work for me. 

```
Traceback (most recent call last):
  File "quickstart.py", line 31, in <module>
    session.like_by_tags(['natgeo'], amount=1)
  File "/home/odroid/InstaPy/instapy/instapy.py", line 961, in like_by_tags
    self.logfolder)
  File "/home/odroid/InstaPy/instapy/comment_util.py", line 39, in comment_image
    rand_comment = emoji.demojize(rand_comment)
  File "/usr/local/lib/python3.5/dist-packages/emoji/core.py", line 76, in demojize
    return get_emoji_regexp().sub(replace, string)
TypeError: cannot use a string pattern on a bytes-like object
```

**NB:** It works like a charm on windows. 

## InstaPy configuration

**Environment :** Odroid C2
**Distribution:** Ubuntu 16.04.4 LTS
**Architecture:** aarch64
**Pyhton version:** Python 3.5.2

**Comment by ghost** [comment]
I still need help. 
I call upon @uluQulu

**Comment by uluQulu** [comment]
hi @ValentinMebarki 
well let's do this quick _switch_
**instead of** encoding _random comment choosing_ line:
```python
rand_comment = (choice(comments).format(username))   #we have removed `.encode('utf-8')`
```
**encode** it in the _demojizing_ line which is right below the prior line:
```python
rand_comment = (emoji.demojize(rand_comment)).encode('utf-8')   #we have added `.encode('utf-8')` here
```
**see** if it works now (_if emojizing line does not make a problem it will work_)

**Comment by ghost** [comment]
hi @uluQulu, thanks for your quick answer
I **did the switch** 
```
rand_comment = (choice(comments).format(username))
rand_comment = (emoji.demojize(rand_comment)).encode('utf-8')
rand_comment = emoji.emojize(rand_comment, use_aliases=True)
```
`logger.info("--> Commented: {}".format(rand_comment))`

and still **gave the same error**
```
Traceback (most recent call last):
  File "quickstart.py", line 31, in <module>
    session.like_by_tags(['natgeo'], amount=1)
  File "/home/odroid/InstaPy/instapy/instapy.py", line 961, in like_by_tags
    self.logfolder)
  File "/home/odroid/InstaPy/instapy/comment_util.py", line 40, in comment_image
    rand_comment = emoji.emojize(rand_comment, use_aliases=True)
  File "/usr/local/lib/python3.5/dist-packages/emoji/core.py", line 52, in emojize
    return pattern.sub(replace, string)
TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by uluQulu** [comment]
@ValentinMebarki 
_emojizing_ line didn't like the encoding, either
let's encode **after it**, similarly
```python
rand_comment = (emoji.emojize(rand_comment, use_aliases=True)).encode('utf-8')
```
**see** if it works now (_don't forget to remove the encoding from the previous test_)

**Comment by ghost** [comment]
@uluQulu 
I **did the change**
```
    rand_comment = (choice(comments).format(username))
    rand_comment = emoji.demojize(rand_comment)
    rand_comment = (emoji.emojize(rand_comment, use_aliases=True)).encode('utf-8')
```
We are one step closer to the solution.
Now I get an **other error** 
```
Traceback (most recent call last):
  File "quickstart.py", line 31, in <module>
    session.like_by_tags(['natgeo'], amount=3)
  File "/home/odroid/InstaPy/instapy/instapy.py", line 961, in like_by_tags
    self.logfolder)
  File "/home/odroid/InstaPy/instapy/comment_util.py", line 50, in comment_image
    "arguments[0].value = '" + rand_comment + " ';", comment_input[0])
TypeError: Can't convert 'bytes' object to str implicitly

```

**EDIT:** 
I tried to add  
```
browser.execute_script(
            "arguments[0].value = '" + rand_comment.decode('utf-8') + " ';", comment_input[0])
```
Then there is no error, but the � still appears after the comment.

**Comment by uluQulu** [comment]
@ValentinMebarki 
try this, in **comment_util.py**, modify `comment_image` function as in below:
find and **replace**
```python
        browser.execute_script(
            "arguments[0].value = '" + rand_comment + " ';", comment_input[0])
        # An extra space is added here and then deleted.
        # This forces the input box to update the reactJS core
        comment_input[0].send_keys("\b")
```
**with**
```python
        comment_input[0].send_keys(rand_comment)
```
**see** if it works now (_also, revert back all modified lines from previous tests at comments above_)

**Comment by ghost** [comment]
@uluQulu 
I **did the change.**

```
comment_input = get_comment_input(browser)

comment_input[0].send_keys(rand_comment)
comment_input = get_comment_input(browser)
```
There is a **new error**
```
Traceback (most recent call last):
  File "quickstart.py", line 31, in <module>
    session.like_by_tags(['natgeo'], amount=3)
  File "/home/odroid/InstaPy/instapy/instapy.py", line 961, in like_by_tags
    self.logfolder)
  File "/home/odroid/InstaPy/instapy/comment_util.py", line 49, in comment_image
    comment_input[0].send_keys(rand_comment)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webelement.py", line 478, in send_keys
    {'text': "".join(keys_to_typing(value)),
TypeError: sequence item 0: expected str instance, int found
```

**Comment by uluQulu** [comment]
@ValentinMebarki 
What are your comments like?

**Comment by ghost** [comment]
@uluQulu 
The error is propagated b**efore the script can comment**.
Here is the **full trace** : 

```
INFO [2018-03-27 10:30:43] [test]  --> Image Liked!
INFO [2018-03-27 10:30:47] [test]  Session ended - 2018-03-27 10:30:47
INFO [2018-03-27 10:30:47] [test]  --------------------


Traceback (most recent call last):
  File "quickstart.py", line 31, in <module>
    session.like_by_tags(['natgeo'], amount=3)
  File "/home/odroid/InstaPy/instapy/instapy.py", line 961, in like_by_tags
    self.logfolder)
  File "/home/odroid/InstaPy/instapy/comment_util.py", line 49, in comment_image
    comment_input[0].send_keys(rand_comment)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webelement.py", line 478, in send_keys
    {'text': "".join(keys_to_typing(value)),
TypeError: sequence item 0: expected str instance, int found

During handling of the above exception, another exception occurred:


Traceback (most recent call last):
  File "quickstart.py", line 35, in <module>
    traceback.print_exc(exc)
  File "/usr/lib/python3.5/traceback.py", line 159, in print_exc
    print_exception(*sys.exc_info(), limit=limit, file=file, chain=chain)
  File "/usr/lib/python3.5/traceback.py", line 100, in print_exception
    type(value), value, tb, limit=limit).format(chain=chain):
  File "/usr/lib/python3.5/traceback.py", line 474, in __init__
    capture_locals=capture_locals)
  File "/usr/lib/python3.5/traceback.py", line 332, in extract
    if limit >= 0:
TypeError: unorderable types: TypeError() >= int()
odroid@odroid64:~/InstaPy$ 
```

**Comment by uluQulu** [comment]
hmmm @ValentinMebarki 
it's very hard to debug virtually :D
but try this,
put `rand_comment` into a string before sends comment
```python
        comment_input[0].send_keys(rand_comment)
```
will be
```python
        comment_input[0].send_keys(str(rand_comment))
```

**Comment by ghost** [comment]
@uluQulu 
I did receive the notification about your answer but it doesn't seem to appear here.
I **made the following change**
`comment_input[0].send_keys(str(rand_comment))`
And ... **no more �** after comments !

**But** the comments are now like this: `b'AmEizing'`  

We can see it in the logs : 
```
INFO [2018-03-27 16:33:31] [test]  --> Image Liked!
INFO [2018-03-27 16:33:33] [test]  --> Commented: b'Wow!'
INFO [2018-03-27 16:33:34] [test]  --> Not following
```

**Comment by uluQulu** [comment]
**After** I sent _that comment_, I did try it _in my own_ and found out that it did not work _so deleted it_, but **Github** _immidiately_ sends email notification **;)**
I think it is definetely **platform specific** problem, that's why _the latest change_ did work for you

_How it does send_ the comment to the target users. **Watch** while bot sends comments in Chrome or open `Posts You've Liked` in IG and check the comments _TO **MAKE** SURE_ whether it sends `b''` bytestring notation there in comments **or not**.
**Else than** there, in **console** output (and **logs**) `b''` bytestring notation is just a notation, nothing more, it happens after _encoding_ to utf-8 and _tends to mean_ that in `b'Wow'`,  `Wow` is a **bytestring** _rather than a_ normal **string**. I previously **made a way** to remove _that annoying_ `b''` from the console output (_and then from logs_) in #1413. _But if you had_ Python 2, you would not see that `b''` bytestring notation there, it is **completely** _Python 3 specific_.

Does it work, **even if** you send some emojis inside comments, _too_? (e.g. comments=[u'Cool pic 🚙'])

**Comment by ghost** [comment]
@uluQulu 

- I may have misspoke myself. I checked on IG and there is indeed this `b' '`. 
- Concerning the **console output**, I implemented the quick fix you proposed in #1413.
 `logger.info("--> Commented: {}".format(str(rand_comment)[2:-1]))`. Now **it's okay**.
- I sent some **emojis**. They **don't appear.** Their code appears instead. 
`Wow! 😃` gives 
`b'Wow! \xf0\x9f\x98\x83'` on IG.

**Comment by uluQulu** [comment]
it's clear @ValentinMebarki 
emojis do need unicode notation, have you tried `u'Wow! 😃'` instead of plain `'Wow! 😃'`

**Comment by ghost** [comment]
@uluQulu 
I tried  `session.set_comments([u'Wow! 😃']). `

But the result on IG is **the same** : 
`b'Wow! \xf0\x9f\x98\x83'`

**Comment by uluQulu** [comment]
Hey @ValentinMebarki 
**With my very less information** on web stuff it's _really_ diffucult issue but I have **hints** about the problem
**I think** the problem is with `encoding`, _specifically_, **some program** in your system _which is responsible for_ `encoding` part, **that's why you do** not see _the same problem in Windows_. 
**And** to be _more precise_ the problem _may lie_ in **ASCII** `encoding`, **cos** _Windows is very ASCII friendly_ while you linux box **may not** and the '\b' has _a special meaning_ for the **ASCII** `encoding` (_removes_ one character backwards, _like in_ JS) and **funnily** `/b` _is also_ **a control key** for ReactJS _which is used by IG_ and **as a result**, although, _I have no any robust understanding_, I _think_ that **diamond question mark** is a result of your _default_ `encoder`
_**We** have_ almost **proved** it by using _another method_ above _rather than_ `send_keys('\b')` and _it **did not** **post** that symbol_. **But** that method _does not work_ with **emojis** (_cos Chrome only supports BMP and we need SMP support for emojis, Firefox supports it, though_), **that's why** we must use the **currently default method** in our _main_ repo _(set value within JS **and** send a control key to commit)_ but **meanwhile** we must **solve** that _conflicting_ error _happened to many_, the diamond question mark....

I don't **know** how well it will **work**, _but try_ this:
**first**, revert back all the changes we did in **comment_util.py**, and use the official file in the main repo
**1**. in **comment_util.py**, import selenium `Keys` near other imports at top of the file
```python
from selenium.common.exceptions import WebDriverException
from selenium.webdriver.common.keys import Keys    ##you will add this new line
```

**2**. then **replace**
```python
comment_input[0].send_keys("\b")
```
**with**
```python
comment_input[0].send_keys(Keys.BACKSPACE) 
```

**Comment by ghost** [comment]
@uluQulu 
**Oh Lord, it works !!!** _I'm dancing on my chair right now._

I reverted back all the changes.
Then I applied this fix. 

I think the issue of the � symbol affects every **ARM chips**: Raspberry Pi, BeagleBoard, BeagleBone, PandaBoard, Odroid and other single-board computers. It could be interesting to **apply a general fix** concerning those and do a pull request. What do you think ?

**Thank you** sooo much @uluQulu. I really appreciate your help.
The communnity here is lucky to have **a man like you**.

**Comment by uluQulu** [comment]
_You would do the same_ @ValentinMebarki 🙂 we learn by helping each other 🖐 (_while looking on this issue I grabbed some ReactJS understanding_)
**You** can be **right**, but I have _very less information_ about that topic and I think those `chipsets` does not welcome ASCII at all 😕
**Also**, _the prior method_ offered by @keusta at #1411 worked for some guys, **because** their system **allowed** them to convert strings into bytestrings (_bytestring is safe from such encoding conflicts_) **but this change will** _most probably_ _work for all_, **because** we have **replaced** the **problematic** `\b` with `Keys.BACKSPACE`
**A general** fix would really be highly **helpful** considering the increase in usage of smaller linux boxes with quite **different** encoding styles

> With lots of dedicated geniuses, this community is really great, indeed, I am the lucky one :>

**Comment by keusta** [comment]
Hey Qulu, thx for helping again, the previous fix #1411 does not work anymore for me 👊  and i really dont know why, but i see your answer and looks like im gonna try again really soon ! see ya dude
